sdk/monitor/azure-monitor-opentelemetry/setup.py

```diff
@@ -0,0 +1,107 @@
+
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "azure-monitor-opentelemetry"
+PACKAGE_PPRINT_NAME = "Azure Monitor Opentelemetry Distro"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+
+# azure v0.x is not compatible with this package
+# azure v0.x used to have a __version__ attribute (newer versions don't)
+try:
+    import azure
+
+    try:
+        ver = azure.__version__
+        raise Exception(
+            "This package is incompatible with azure=={}. ".format(ver)
+            + 'Uninstall it with "pip uninstall azure".'
+        )
+    except AttributeError:
+        pass
+except ImportError:
+    pass
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(
+        r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE
+    ).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {} Client Library for Python".format(
+        PACKAGE_PPRINT_NAME
+    ),
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="ascl@microsoft.com",
+    url="https://github.com/microsoft/ApplicationInsights-Python/tree/main/azure-monitor-opentelemetry",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
+    packages=find_packages(
+        exclude=[
+            "tests",
+            "samples",
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+            "azure.monitor",
+        ]
+    ),
+    include_package_data=True,
+    package_data={
+        "pytyped": ["py.typed"],
+    },
+    python_requires=">=3.7",
+    install_requires=[
+        "azure-monitor-opentelemetry-exporter>=1.0.0b13",
+        "opentelemetry-instrumentation~=0.38b0",
+        "opentelemetry-instrumentation-django~=0.38b0",
+        "opentelemetry-instrumentation-fastapi~=0.38b0",
+        "opentelemetry-instrumentation-flask~=0.38b0",
+        "opentelemetry-instrumentation-psycopg2~=0.38b0",
+        "opentelemetry-instrumentation-requests~=0.38b0",
+        "opentelemetry-instrumentation-urllib~=0.38b0",
+        "opentelemetry-instrumentation-urllib3~=0.38b0",
+        "opentelemetry-api==1.17.0",
+        "opentelemetry-sdk==1.17.0",
+    ],
+    entry_points={
```

Comment:
@annatisch These entry points are required for OpenTelemetry auto-instrumentation. In particular, is used in our Rp Integration for Azure App Service. Note that this package is a "distro." But, "distro" is also an OpenTelemetry term for classes that are utilized in the OpenTelemetry auto-instrumentation flow. Our classes in the auto-instrumentation folder are a part of this.

Reply:
Is it a requirement that all "distro" packages have entry points?
If so - there should be a common naming convention? "Configurator" is not a word - is that some kind of otel concept that's the name of the entrypoint for other distros?
I'm not sure what you mean by "our RP Integration for Azure App Service" - does that mean we're using it internally?

Reply:
Yes, here our some code references for OTEL auto instrumentation:
[distro](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/511709802466a751786047b7d98c2eb84801b34f/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py#LL35C51-L35C51)
[configurator](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/511709802466a751786047b7d98c2eb84801b34f/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py#L96)
As for app service, yes. Our Auto-Attach option for App Service is build on top of the distro package. It is currently in private preview looking for customers.